File: Year 2/Semester 4/Programming and Projection Enironments/Week 4/C#FourthWeek/SaptamanaAPatra/repository/interfaces/IRepository.cs
using SaptamanaAPatra.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SaptamanaAPatra.repository.interfaces
{
    internal interface IRepository<ID, E> where E : Entity<ID>
    {
        List<E> GetAll();
        E FindById(ID id);
        bool Add(E entity);
        bool Delete(ID id);
        bool Update(E entity);
    }
}

File: Year 2/Semester 4/Programming and Projection Enironments/Exam preparation/Joc gropi/Martin - examen/Persistence/src/main/java/app/rest/persistance/repository/IRepository.java
package app.rest.persistance.repository;

import domain.Entity;

import java.io.Serializable;

public interface IRepository<ID extends Serializable, E extends Entity<ID>>  {
    /**
     *
     * @param id of type ID (template)
     * @return found entity or null
     */
    E findById(ID id);
    Iterable<E> getAll();
    boolean add(E entity);

    boolean update(E entity);
}
